src/Components/Aspire.MongoDB.Driver/AspireMongoDBDriverExtensions.cs

```diff
@@ -0,0 +1,233 @@
+
+        builder.AddMongoClient(
+            settings,
+            connectionName,
+            configurationSectionName,
+            configureClientSettings,
+            serviceKey);
+
+        if (settings.Tracing)
+        {
+            builder.Services
+                .AddOpenTelemetry()
+                .WithTracing(tracer => tracer.AddSource(ActivityNameSource));
+        }
+
+        builder.AddMongoDatabase(settings.ConnectionString, serviceKey);
+        builder.AddHealthCheck(
+            serviceKey is null ? "MongoDB.Driver" : $"MongoDB.Driver_{connectionName}",
+            settings);
+    }
+
+    private static void AddMongoClient(
+        this IHostApplicationBuilder builder,
+        MongoDBSettings mongoDbSettings,
+        string connectionName,
+        string configurationSectionName,
+        Action<MongoClientSettings>? configureClientSettings,
+        object? serviceKey)
+    {
+        if (string.IsNullOrWhiteSpace(serviceKey as string))
+        {
+            builder
+                .Services
+                .AddSingleton<IMongoClient>(sp => sp.CreateMongoClient(connectionName, configurationSectionName, mongoDbSettings, configureClientSettings));
+            return;
+        }
+
+        builder
+            .Services
+            .AddKeyedSingleton<IMongoClient>(serviceKey, (sp, _) => sp.CreateMongoClient(connectionName, configurationSectionName, mongoDbSettings, configureClientSettings));
+    }
+
+    private static void AddMongoDatabase(
+        this IHostApplicationBuilder builder,
+        string? connectionString,
+        object? serviceKey = null)
+    {
+        if (string.IsNullOrWhiteSpace(connectionString))
+        {
+            return;
+        }
+
+        var mongoUrl = MongoUrl.Create(connectionString);
+
+        if (string.IsNullOrWhiteSpace(mongoUrl.DatabaseName))
+        {
+            return;
+        }
+
+        if (string.IsNullOrWhiteSpace(serviceKey as string))
+        {
+            builder.Services.AddSingleton<IMongoDatabase>(provider =>
+            {
+                return provider
+                    .GetRequiredService<IMongoClient>()
+                    .GetDatabase(mongoUrl.DatabaseName);
+            });
+
+            return;
+        }
+
+        builder.Services.AddKeyedSingleton<IMongoDatabase>(serviceKey, (provider, _) =>
+        {
+            return provider
+                .GetRequiredKeyedService<IMongoClient>(serviceKey)
+                .GetDatabase(mongoUrl.DatabaseName);
+        });
+    }
+
+    private static void AddHealthCheck(
+        this IHostApplicationBuilder builder,
+        string healthCheckName,
+        MongoDBSettings settings)
+    {
+        if (!settings.HealthChecks || string.IsNullOrWhiteSpace(settings.ConnectionString))
+        {
+            return;
+        }
+
+        builder.TryAddHealthCheck(
+            healthCheckName,
+            healthCheck => healthCheck.AddMongoDb(
+                settings.ConnectionString,
```

Comment:
Ideally this should use the same IMongoClient instance that is registered in DI. See https://github.com/dotnet/aspire/blob/main/src/Components/README.md#health-checks. 

> Consider whether the Health Check should reuse the same client object registered in DI by the component or not. Reusing the same client object has the advantages of getting the same configuration, logging, etc during the health check.

Reply:
The **AspNetCore.HealthChecks.MongoDb** does not support to pass an IMongoClient instance.
It is only possible to pass the connection string or the MongoClientSettings

Reply:
I've created a new PR to add support for receiving a IMongoClient instance when configuring the healthcheck

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/2121